src/apps/testapps/testH3IndexFat.c

```diff
@@ -1,148 +0,0 @@
-/*
- * Copyright 2017 Uber Technologies, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *         http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-/** @file
- * @brief tests H3 functions for manipulating H3 indexes
- *
- *  usage: `testH3IndexFat`
- */
-
-#include "h3Index.h"
-#include "h3IndexFat.h"
-#include "test.h"
-#include "utility.h"
-
-BEGIN_TESTS(h3IndexFat);
-
-TEST(h3FatIsValidWithMode) {
-    for (int i = 0; i <= 0xf; i++) {
-        H3IndexFat hf;
-        setH3IndexFat(&hf, 0, 0, 0);
-        hf.mode = i;
-        char failureMessage[BUFF_SIZE];
-        sprintf(failureMessage, "h3FatIsValid failed on mode %d", i);
-        t_assert(!h3FatIsValid(&hf) || i == 1, failureMessage);
-    }
-}
-
-TEST(h3FatNegativeResInvalid) {
-    H3IndexFat hfNegativeRes;
-    setH3IndexFat(&hfNegativeRes, -1, 0, 0);
-    t_assert(!h3FatIsValid(&hfNegativeRes),
-             "h3FatIsValid failed on negative resolution");
-}
-
-TEST(h3FatLargeResolutionInvalid) {
-    H3IndexFat hfLargeRes;
-    setH3IndexFat(&hfLargeRes, 15, 0, 0);
-    // Can't set resolution to 16 using setH3IndexFat
-    // since it would corrupt the stack.
-    hfLargeRes.res = 16;
-    t_assert(!h3FatIsValid(&hfLargeRes),
-             "h3FatIsValid failed on large resolution");
-}
-
-TEST(h3FatUnusedDigitInvalid) {
-    H3IndexFat hf;
-    initH3IndexFat(&hf, 0);
-    hf.baseCell = 1;
-    hf.index[0] = 0;
-    t_assert(!h3FatIsValid(&hf),
-             "h3FatIsValid failed on invalid index digit (0)");
-
-    hf.index[0] = 7;
-    hf.index[14] = 2;
-    t_assert(!h3FatIsValid(&hf),
-             "h3FatIsValid failed on invalid index digit (14)");
-
-    hf.res = 1;
-    hf.index[14] = 7;
-    hf.index[1] = 1;
-    t_assert(!h3FatIsValid(&hf),
-             "h3FatIsValid failed on invalid index digit (1)");
-}
-
-TEST(h3FatBadDigitInvalid) {
```

Comment:
I don't think this is tested (would look like res=1 but the digits are all 7)

Reply:
Skipping the negative test (that can't exist, now) and porting the digit 7 test.